optuna/_transform.py

```diff
@@ -341,7 +341,9 @@ def _untransform_numerical_param(
             else:
                 param = min(trans_param, numpy.nextafter(d.high, d.high - 1))
     elif isinstance(d, IntUniformDistribution):
-        param = int(numpy.round((trans_param - d.low) / d.step) * d.step + d.low)
+        param = int(
+            min(max(numpy.round((trans_param - d.low) / d.step) * d.step + d.low, d.low), d.high)
+        )
```

Comment:
I try the following, and pass the unit test and mypy.
```suggestion
        param = int(numpy.clip(numpy.round((trans_param - d.low) / d.step) * d.step + d.low, d.low, d.high))
```

Reply:
Brilliant! Thanks!